packages/build/src/plugins_core/blobs_upload/index.ts

```diff
@@ -48,34 +49,35 @@ const coreStep: CoreStepFunction = async function ({
     return {}
   }
 
-  // If using the deploy config API, configure the store to use the region that
-  // was configured for the deploy.
-  if (!blobs.isLegacyDirectory) {
+  // If using the deploy config API or the Frameworks API, configure the store
+  // to use the region that was configured for the deploy. We don't do it for
+  // the legacy file-based upload API since that would be a breaking change.
+  if (blobs.apiVersion > 1) {
     storeOpts.experimentalRegion = 'auto'
   }
 
   const blobStore = getDeployStore(storeOpts)
-  const keys = await getKeysToUpload(blobs.directory)
+  const blobsToUpload = blobs.apiVersion >= 3 ? await getBlobs(blobs.directory) : await getKeysToUpload(blobs.directory)
```

Comment:
`blobsToUpload` is now an array of objects with the following properties:

- `key`: The blob key
- `contentPath`: The absolute path to the blob content file
- `metadataPath`: The absolute path to the blob metadata file (doesn't mean that one exists)